sys/linux/dev_kvm.txt

```diff
@@ -190,9 +191,11 @@ syz_kvm_setup_cpu$ppc64(fd fd_kvmvm, cpufd fd_kvmcpu, usermem vma[24], text ptr[
 
 resource kvm_run_ptr[int64]
 define KVM_RUN_SIZE	sizeof(struct kvm_run)
-mmap$KVM_VCPU(addr vma, len const[KVM_RUN_SIZE], prot flags[mmap_prot], flags flags[mmap_flags], cpufd fd_kvmcpu, offset const[0]) kvm_run_ptr
+mmap$KVM_VCPU(addr vma, len vcpu_mmap_size, prot flags[mmap_prot], flags flags[mmap_flags], cpufd fd_kvmcpu, offset const[0]) kvm_run_ptr
 _ = __NR_mmap2
 
+syz_kvm_run_expect_uexit$arm64(run kvm_run_ptr, expect intptr)
```

Comment:
Does everything work if you add `(disabed)` attribute?
We generally don't have test-only syscalls for linux, at least we need to disable during fuzzing. But IIRC disabled syscalls are allowed in the manual tests.

... thinking of this more: tests also used as seeds. And we will have all tests sprinkled with test-only syscalls...

Hard to design tests while we don't have a representative set of things (api calls) we want to test.
Is it possible to observe effects of all api calls using other "production" api calls, or normal syscalls?

Reply:
I don't think having test-only syscalls in the seeds is a big problem.
If they don't produce coverage, syzkaller will soon learn to not generate them.
If they do (e.g. they implement a KVM_RUN loop), it would be nice to have them, because they'll offer richer semantics for syzlang programs.

> Is it possible to observe effects of all api calls using other "production" api calls, or normal syscalls?
We might need a way to make assertions on values in memory (e.g. `struct kvm_run` or the pages we passed to `syz_kvm_setup_cpu$arm64`), and that does not require syscalls at all.
We could introduce some special prefix (e.g. `syz_check_`) for syscalls that should never be used in program generation. But we could as well stick with the current design and wait till it becomes a problem.